### PR TITLE
Add hint willReadFrequently for canvas

### PR DESCRIPTION
**Pull request by axinging**
Add hint willReadFrequently for canvas

This change has performance impact on cpu and wasm backend. For
getImageData involved cases, set willReadFrequently to true may
improve performance, for example, for input image 125x125 to 513x513,
the measured time on getImageData is ~ 20ms when willReadFrequently: false, 
and 6~10 ms when willReadFrequently: true.

The root cause is, getImageData runs on different hardware (CPU, GPU).
See comments here: https://bugs.chromium.org/p/chromium/issues/detail?id=1239467

Use tf.env().set('CANVAS2D_WILL_READ_FREQUENTLY', true) to turn on this feature.

Spec: https://github.com/fserb/canvas2D/blob/master/spec/will-read-frequently.md
Bug: https://github.com/tensorflow/tfjs/issues/4227, https://github.com/tensorflow/tfjs/issues/4218

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6445)

### CODE REVIEW

**Comment by gyagp** [comment]
Why not set willReadFrequently to true unconditionally for all the backends?

**Comment by axinging** [comment]
> Why not set willReadFrequently to true unconditionally for all the backends?
From https://github.com/fserb/canvas2D/blob/master/spec/will-read-frequently.md, it says the call path of getImageData  was first "heuristics " then turned to a explicit hint willReadFrequently, because this has different performance:
"
Historically, browsers have tried heuristics to decide if a canvas is being read frequently or not, to switch code paths.
This is unreliable, complex and brittle. To prevent this, we propose a hint option that will allow developers to tell the browser that this is a canvas that rely on getImageData performance.
"
So in this PR, I tried to align it with getImageData.

**Comment by axinging** [comment]
@qjia7 @gyagp @xhcao @haoyunfeix PTAL

**Review by qjia7 — APPROVED** [review]
Awesome!

**Comment by gyagp** [comment]
@pyu10055 @lina128 This PR supposes to fix the long-standing fromPixels() perf issue, but we're not sure about its side effect. So an env variable is provided now (off by default) and we'd hope developers may have some early tests. If the side effect is minor, it can be turned on by default for CPU and WASM backends.

**Review by lina128 — APPROVED** [review]
Thank you! LGTM!

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6445#-:-N3WWVqZBIoCOIUkx9K7:bebeqq2)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @pyu10055)

**Review by pyu10055 — APPROVED** [review]
Thank you for the investigation and fixing this issue!

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6445#-:-N3WWzBc87zvBpp7y3um:bw102kg)* status: :shipit: complete! 3 of 1 approvals obtained

File: tfjs-core/src/ops/browser.ts [hunk 1]
```diff
@@ -155,7 +155,11 @@ function fromPixels_(
               'Reason: OffscreenCanvas Context2D rendering is not supported.');
         }
       } else {
-        fromPixels2DContext = document.createElement('canvas').getContext('2d');
+        const willReadFrequently =
+            env().getBool('CANVAS2D_WILL_READ_FREQUENTLY');
+        const options = willReadFrequently ? {willReadFrequently} : {};
+        fromPixels2DContext = document.createElement('canvas').getContext(
```
Comment (qjia7) [inline tfjs-core/src/ops/browser.ts:161]: nit: It seems that you can directly write it like below and remove options. Tested [here](https://github.com/tensorflow/tfjs/pull/6486).
```
        fromPixels2DContext =
            document.createElement('canvas').getContext(
                '2d', {willReadFrequently}) as CanvasRenderingContext2D;

**Pull request merged**